### ISSUE TITLE
OpenCL.dll is missing on systems with no GPU's installed

### ISSUE THREAD

**Issue by rlincoln24** [MultiPoolMiner/MultiPoolMiner#862]
Albeit a very small use case I have a dedicated server with xeon processors running VMWare esxi and I have a vm that I want to run MultiPoolMiner on for CPU Mining only there is no GPU installed at all and thus no OpenCL.dll was ever installed.

When I ran the start.bat it continuously gave me errors about OpenCL.dll missing. I tried copying it from a system that had it installed to no avail.

I was getting the following errors:

1.
`PS>TerminatingError(Invoke-Expression): "Exception calling "GetPlatformIDs" with "0" argument(s): "Unable to load DLL 'opencl.dll': The specified module could not be found. (Exception from HRESULT: 0x8007007E)""`

2.
```
PS>TerminatingError(Compare-Object): "Cannot bind argument to parameter 'DifferenceObject' because it is null."
Compare-Object : Cannot bind argument to parameter 'DifferenceObject' because it is null.
```

3.
`PS>TerminatingError(Invoke-Expression): "Exception calling "GetPlatformIDs" with "0" argument(s): "OpenCl error -1001: -1001.""`

I ended up installing the following from [here](https://software.intel.com/en-us/articles/opencl-drivers)  
 the OpenCL™ Runtime 16.1.2 for Intel® Core™ and Intel® Xeon® Processors for Windows* (64-bit & 32-bit) under the OpenCL™ Runtime for Intel® Core™ and Intel® Xeon® Processors section. Once I installed that everything worked perfectly. There are versions for Linux on there as well but I don't know anything about those and if they solve this problem. My VM is running server 2012 R2 Datacenter edition.

Thanks for all the hardwork aaron! You can close this issue I mostly wanted to open this for documentation for other people searching the issues and struggling with OpenCL.dll issues.

**Comment by platinum4** [comment]
This is an array it gathers to figure out if it needs to put AMD on platform 1 and NVIDIA on platform 0 so if you had no GPUs then yes by default this wouldn't likely mine but you could also just edit that gather function out of the MultiPoolMiner.ps1

This is niche use-case but thanks.  I imagine you had the Xeon booted through remote?

Also, you needed to only specify CPU in -platform switch in the .bat file, if you left amd,nvidia,cpu then it is going to look for GPUs.

**Comment by rlincoln24** [comment]
I did specify only CPU in the -platform command line but ran into this still.

I am not 100% sure what "had the Xeon booted through remote" means but if it means was I remoted into it using Remote desktop or some other remote software the answer is no. I was RDP'd into it performing the setup and the subsequent installs and then attempting to load it via vSphere console window.

I agree this is a super niche case but thought i'd put it out there in case someone ran into something similar and wanted to fix it without going through and editing the miner's core code.
